src/spatio_temporal_voxel_grid.cpp

```diff
@@ -237,19 +237,20 @@ void SpatioTemporalVoxelGrid::TemporalClearAndGenerateCostmap(                \
       }
     }
     // if here, we can add to costmap and PC2
-    PopulateCostmapAndPointcloud(pt_index);
+    PopulateCostmapAndPointcloud(pt_index, true);
   }
 }
 
 /*****************************************************************************/
-void SpatioTemporalVoxelGrid::PopulateCostmapAndPointcloud(const \
-                                                            openvdb::Coord& pt)
+void SpatioTemporalVoxelGrid::PopulateCostmapAndPointcloud(                   \
+                                                   const openvdb::Coord& pt,  \
+                                                   const bool& pub_every_voxel)
 /*****************************************************************************/
 {
   // add pt to the pointcloud and costmap
   openvdb::Vec3d pose_world = this->IndexToWorld(pt);
 
-  if (_pub_voxels)
+  if (_pub_voxels && pub_every_voxel)
```

Comment:
This doesn't make sense - you'd now be inserting it twice if its new. Once here and once below in the `else` statement. Please look over the functions after you prototype something to make sure it makes sense